src/uct/tcp/sockcm/sockcm_ep.c

```diff
@@ -14,26 +14,267 @@
         } \
     } while (0)
 
+ucs_status_t uct_sockcm_ep_set_sock_id(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    ucs_status_t status;
+    struct sockaddr *dest_addr = NULL;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    ep->sock_id_ctx = ucs_malloc(sizeof(*ep->sock_id_ctx), "client sock_id_ctx");
+    if (ep->sock_id_ctx == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
+
+    dest_addr = (struct sockaddr *) &(ep->remote_addr);
+
+    status = ucs_socket_create(dest_addr->sa_family, SOCK_STREAM,
+                               &ep->sock_id_ctx->sock_id);
+    if (status != UCS_OK) {
+        ucs_debug("unable to create client socket for sockcm");
+        goto out_free;
+    }
+
+    ucs_list_add_tail(&iface->used_sock_ids_list, &ep->sock_id_ctx->list);
+    ucs_debug("ep %p, new sock_id %d", ep, ep->sock_id_ctx->sock_id);
+    status = UCS_OK;
+    goto out;
+
+out_free:
+    ucs_free(ep->sock_id_ctx);
+out:
+    UCS_ASYNC_UNBLOCK(iface->super.worker->async);
+    return status;
+}
+
+ucs_status_t uct_sockcm_ep_send_client_info(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    uct_sockcm_conn_param_t conn_param;
+    uct_sockcm_priv_data_hdr_t *hdr;
+    ssize_t transfer_len = 0;
+    ssize_t sent_len = 0;
+
+    memset(&conn_param, 0, sizeof(uct_sockcm_conn_param_t));
+    hdr = &conn_param.hdr;
+
+    /* pack worker address into private data */
+    hdr->length = ep->pack_cb(ep->pack_cb_arg, "DEV_NAME_NULL",
```

Comment:
Agreed. Will address this.

Reply:
I just tested with valid IPs (not INADDR_ANY) at the listener and the current PR seems to handle this. 

That said, it seems like the following code path (in src/ucp/wireup/wireup_ep.c) isn't stressed for the tests that I'm running:
```
525     if (conn_priv_len > attrs->max_conn_priv) {
526 
527         /* since the full worker address is too large to fit into the trasnport's
528          * private data, try to pack sockaddr aux tls to pass in the address */
529         status = ucp_wireup_ep_pack_sockaddr_aux_tls(worker, dev_name, &tl_bitmap,
530                                                      &rsc_address, &address_length);
531         if (status != UCS_OK) {                      
532             goto err_free_address;
533         }   
534 
```
Specifically, we're not making a call to `ucp_wireup_ep_pack_sockaddr_aux_tls`. It seems that just an auxiliary transport capable of active message bcopy is packed when the size of the addresses packed is beyond the limit (`attrs->max_conn_priv`) suggested in line 525. For such a case it appears I have to specify a valid uct device name and not DEV_NAME_NULL. Since sockcm depends on sockets functionality, is it safe to give `tcp` uct device name (because its iface claims UCT_IFACE_FLAG_AM_BCOPY support)? Is it guaranteed that uct tcp am_bcopy is already up when `ucp_wireup_ep_pack_sockaddr_aux_tls` call is made? If not, can you suggest which device name to use such that it works in both intra-node and inter-node cases? Any other thoughts?

Reply:
Copied some code from `tcp` uct to address this here https://github.com/openucx/ucx/pull/3570/commits/450a0992dea4a4d5bbfceffecac96167a1e6806d 

Will this work?

Reply:
the sockcm transport shouldn't rely on another transport (tcp) to be up, it should have it's own ability to get the required device name.